src/days-2-4-adv/room.py

```diff
@@ -1,2 +1,42 @@
 # Implement a class to hold room information. This should have name and
-# description attributes.
\ No newline at end of file
+# description attributes.
+
+class Room():
+    def __init__(self, location, message):
+        self.location = location
+        self.message = message
+        self.n_to = None
+        self.e_to = None
+        self.s_to = None
+        self.w_to = None
+
+    def __repr__(self):
+        return f"location: {self.location}, message: {self.message}"
+
+    def direction(self, direction):
+        if direction == "n":
+            return self.n_to
+        elif direction == "s":
+            return self.s_to
+        elif direction == "e":
+            return self.e_to
+        elif direction == "w":
+            return self.w_to
+        else:
+            return "None"
+
+    def connectRooms(self, newRoom, direction):
+        if direction == "n":
+            self.n_to = newRoom
+            newRoom.s_to = self
+        elif direction == "s":
+            self.s_to = newRoom
+            newRoom.s_to = self
+        elif direction == "e":
+            self.e_to = newRoom
+            newRoom.e_to = self
+        elif direction == "w":
+            self.w_to = newRoom
+            newRoom.w_to = self
+        else:
+            print("Sorry, you can't move in that direction, please enter n, s, e, w")
```

Comment:
I like how you are putting functionality related to a class of objects in that class' methods. 😍